sparse_strips/vello_hybrid/src/image_cache.rs

```diff
@@ -80,6 +80,14 @@ impl ImageCache {
         self.slots.get(id.as_u32() as usize)?.as_ref()
     }
 
+    /// Check if an image is in the cache
+    pub(crate) fn contains(&self, id: ImageId) -> bool {
```

Comment:
I think if we wanted to make this more resilient we could add a generation to `ImageId`:

```rs
let id1 = renderer.upload_image(...);
renderer.destroy_image(id1);
let id2 = renderer.upload_image(...);
image_cache.contains(id1); // returns true because id2 uses the same u32 as id1 due to the free list
```

If we incremented a generation whenever we destroy an image, we'd be protected.

More practically, this would also protect from double destroys:

```rs
let id1 = renderer.upload_image(...);
renderer.destroy_image(id1);
let id2 = renderer.upload_image(...);
renderer.destroy_image(id1); // destroys id2
```

This also overcomes that comment in `schedule.rs`